[GTK][WPE] Follow-up to HyperlinkAuditingEnabled deprecation in 295103@main

aperezdc opened the pull request:
#### 88869437e3cbd20e6b5def8a0e5d70d3f92ca5b9
<pre>
[GTK][WPE] Follow-up to HyperlinkAuditingEnabled deprecation in 295103@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=293245">https://bugs.webkit.org/show_bug.cgi?id=293245</a>

Reviewed by Michael Catanzaro.

* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webkit_settings_get_enable_hyperlink_auditing): Make the returned value
consistent with the default property value (TRUE), and do not warn on
getter usage.
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitSettings.cpp:
(testWebKitSettings): Bring back the API test, to ensure that the setter
does not change the property after all.

Canonical link: <a href="https://commits.webkit.org/295155@main">https://commits.webkit.org/295155@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4b355ab00b5b92af27413333910b078098145d0e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/104249 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/23953 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/14319 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/109449 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/54917 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/106289 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/24323 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/32498 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/79177 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/107255 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18900 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/94069 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/59505 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/18708 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/12119 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/54277 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/88409 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/12179 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/111831 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/31405 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/23147 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/88198 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/31770 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/90276 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/87861 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32752 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/10509 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/25869 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16925 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/31334 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/36647 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/31128 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/34464 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/32688 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
```diff
@@ -2154,11 +2154,9 @@ void webkit_settings_set_javascript_can_open_windows_automatically(WebKitSetting
 {
-    g_return_val_if_fail(WEBKIT_IS_SETTINGS(settings), FALSE);
-
-    g_warning("webkit_settings_get_enable_hyperlink_auditing is deprecated and always returns FALSE.");
```
If this is not desired you should also remove it from webkit_settings_get_enable_dns_prefetching. Or is it still desired if it returns false?

aperezdc replied:
My understanding is that we do not have a strongly agreed policy on this regard for the GTK/WPE ports, and that the idea has been to warn when trying to set the option to its non-default value. Not super important, because this is unlikely to spam the logs&mdash;typically applications are not all the time changing settings after all&mdash;but still nice to try and be consistent.

I would have commented on the patch that removed the DNS prefetching option, but I managed to miss it when it landed 😇. Maybe I'll send a follow up when I have some spare cycles for that.

mcatanzaro replied:
I was thinking about just deleting all of the warnings, because they're not very useful.

But I think they are OK as long as we don't warn when setting to default value, which happens during WebKitSettings construction as every one of these setters gets called.

annevk commented on Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitSettings.cpp:
```diff
@@ -89,6 +89,13 @@ static void testWebKitSettings(Test*, gconstpointer)
+    g_assert_true(webkit_settings_get_enable_hyperlink_auditing(settings));
+    webkit_settings_set_enable_hyperlink_auditing(settings, FALSE);
+    g_assert_true(webkit_settings_get_enable_hyperlink_auditing(settings));
+    ALLOW_DEPRECATED_DECLARATIONS_END
```
If this is desired you should probably also restore the test for webkit_settings_get_enable_dns_prefetching.

aperezdc replied:
Done in https://github.com/WebKit/WebKit/pull/47405 🙃

aperezdc committed:
[GTK][WPE] Follow-up to HyperlinkAuditingEnabled deprecation in 295103@main
https://bugs.webkit.org/show_bug.cgi?id=293245

Reviewed by Michael Catanzaro.

* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webkit_settings_get_enable_hyperlink_auditing): Make the returned value
consistent with the default property value (TRUE), and do not warn on
getter usage.
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitSettings.cpp:
(testWebKitSettings): Bring back the API test, to ensure that the setter
does not change the property after all.

Canonical link: https://commits.webkit.org/295155@main
